### PR TITLE
[14.0][MIG] sale_layout_category_hide_detail

### PR DESCRIPTION
**Pull request by rousseldenis**
Added:

- [x] Able to hide section line subtotal

### CODE REVIEW

**Comment by rousseldenis** [comment]
@ernestotejeda This is an interesting widget.

But what do you think about making it more generic (only with the js part), in order to apply it on more apps that have form sections(purchase, ...) ?

**Comment by ernestotejeda** [comment]
> But what do you think about making it more generic ...

I think that's a good idea. what do you think @pedrobaeza ?

**Comment by pedrobaeza** [comment]
As the widget has been moved from sale to account during previous version (on the initial version of this module it wasn't), we can have an `account_layout_category_hide_detail` with account.move + widget stuff, and then add the sale one on top of it, as well as the purchase can be created easily.

I agree it can be a good move.

File: sale_layout_category_hide_detail/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Sale layout category hide detail",
+    "summary": "Hide details for sections in sale orders and invoices for "
+    "reports and customer portal",
+    "version": "14.0.1.0.0",
+    "category": "Sales Management",
+    "website": "https://github.com/OCA/sale-reporting",
+    "author": "Tecnativa, " "Odoo Community Association (OCA)",
```
Comment (ernestotejeda) [inline sale_layout_category_hide_detail/__manifest__.py:11]:
```suggestion
    "author": "Tecnativa, Odoo Community Association (OCA)",
```

**Review by ernestotejeda — COMMENTED** [review]
Funcional review 👍🏼 

- I think readme should be improved to add explanation about 'Show line amount' option 
- You could take advantaje of the migration to adapt javascript code to ES, what do you think?

**Review by angelmoya — APPROVED** [review]

**Review by angelmoya — COMMENTED** [review]
Functional review:

With "line amount hiden", pdf works, but website show details.

**Comment by rousseldenis** [comment]
> Functional review:
> 
> With "line amount hiden", pdf works, but website show details.

Indeed. But after check, it was present in previous versions. I'll try to fix it

**Comment by rousseldenis** [comment]
@angelmoya Could you try now ?

**Review by ernestotejeda — COMMENTED** [review]
Code review: 👍🏼 
Functional review: 👍🏼 

I remember when I migrated the module to v12 @pedrobaeza suggested me to have the section name in the subtotal: https://github.com/OCA/sale-reporting/pull/70#pullrequestreview-259434580.
@pedrobaeza Is there a specific reason to do that or we  can keep the total without the section name as Rousseldenis has done now for v14?

**Comment by pedrobaeza** [comment]
Yes, it's on purpose, and specified by our customer, as when having very big sections, you don't know which section you are putting the subtotal for. Can you please restore it, @rousseldenis, or add a configuration company parameter for specifying the behavior?

**Comment by rousseldenis** [comment]
> Yes, it's on purpose, and specified by our customer, as when having very big sections, you don't know which section you are putting the subtotal for. Can you please restore it, @rousseldenis, or add a configuration company parameter for specifying the behavior?

Ok, try to do that. I understand the purpose. But IMHO the presentation is a little bit dirty. I'm not expert in UI. Keep in mind section name can be long and subtotal string can be 'lost in space'...

**Comment by pedrobaeza** [comment]
One thing we can do is to put such text in a separate element for not moving `Subtotal` text.

**Comment by elvise** [comment]
> One thing we can do is to put such text in a separate element for not moving `Subtotal` text.

Hi @rousseldenis great PR!!!
any plan for get it done ?

**Comment by elvise** [comment]
@rousseldenis can we add an option for page break ?

**Comment by angelmoya** [comment]
I try Todo page break in this version, and this is not easy to do, beacuse sections are inside table, and div page break do not work inside a table.

To do it, it is required to redefine table and create a table for section, and it's not possible to do inheriting views.

Maybe you can do other way, or maybe page break can work inside a table...

**Comment by francesco-ooops** [comment]
Hi @rousseldenis , any plans for this one?

**Comment by rousseldenis** [comment]
> Hi @rousseldenis , any plans for this one?

I'll try to get back here today or tomorrow

**Comment by angelmoya** [comment]
Functional revision not OK (tested on runbot), on posted invoice clicking on button "Switch to: detail hiden" does visible the "paid" ribbon

image

**Comment by rousseldenis** [comment]
> Functional revision not OK (tested on runbot), on posted invoice clicking on button "Switch to: detail hiden" does visible the "paid" ribbon
> 
> image

Thanks for the review but this module does not take care about that ribbon. Only lines.

**Comment by ernestotejeda** [comment]
ping @rousseldenis :)

**Comment by jeroenev** [comment]
> Thanks for the review but this module does not take care about that ribbon. Only lines.

If the module breaks default Odoo functionality, I think it's relevant to fix this before finishing this PR
because it does seem like that has an unintended impact on the payment banner

**Comment by ao-landoo** [comment]
+1 to the bug, it is marking the invoice as paid when it shouldn't. I've been looking for the fix for a few hours but I cannot find the problem's solution...

File: sale_layout_category_hide_detail/static/src/js/boolean_fa_icon_widget.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        /**
+         * Toggle value
+         *
+         * @private
+         * @param {MouseEvent} event
+         */
+        _toggleValue: function (event) {
+            event.preventDefault();
+            event.stopPropagation();
+            this._setValue(!this.value);
```
Comment (ao-landoo) [inline sale_layout_category_hide_detail/static/src/js/boolean_fa_icon_widget.js:76]: The invoice gets marked as paid when this line is executed

Comment (victoralmau) [reply sale_layout_category_hide_detail/static/src/js/boolean_fa_icon_widget.js:76]: Reviewing this in detail would have a complex solution; however, most importantly, I believe that you should NOT be able to modify anything (use the buttons) if the invoice is not in draft. This change would therefore solve the problem. 
What do you think about it?

Comment (jeroenev) [reply sale_layout_category_hide_detail/static/src/js/boolean_fa_icon_widget.js:76]: If that actually solves the problem then I think it's an acceptable solution

**Comment by victoralmau** [comment]
Superseed by https://github.com/OCA/sale-reporting/pull/170

**Pull request closed without merging**